File: M3 Intermediate DSA/src/L10_Interview_Problems_on_Arrays/Assignment1.java
package L10_Interview_Problems_on_Arrays;

public class Assignment1 {

  public int[] solve(int[] A, int[][] B) {
    int N = A.length;
    int M = B.length;
    int[] ans = new int[M];
    int[] prefixEven = new int[N];

    prefixEven[0] = A[0];

    for (int index = 1; index < N; index++) {
      prefixEven[index] = prefixEven[index - 1] + ((index & 1) == 0 ? A[index] : 0);
    }

    for (int index = 0; index < M; index++) {
      int left = B[index][0];
      int right = B[index][1];
      int sum = prefixEven[right];

      if (left > 0) {
        sum -= prefixEven[left - 1];
      }

      ans[index] = sum;
    }

    return ans;
  }
}

File: M4_Intermediate_DSA/L4_Interview_Problems/assignment1.go
package main

func SolveA1(A string) int {
	onesCount := 0
	N := len(A)
	ans := 0

	for _, ch := range A {
		if ch == '1' {
			onesCount++
		}
	}

	if onesCount == N {
		return N
	}

	for i := range A {
		left := 0
		right := 0

		for j := i - 1; j >= 0; j-- {
			if A[j] == '1' {
				left++
			} else {
				break
			}
		}

		for j := i + 1; j < N; j++ {
			if A[j] == '1' {
				right++
			} else {
				break
			}
		}

		count := left + right
		if count != onesCount {
			count++
		}

		if count > ans {
			ans = count
		}
	}

	return ans
}
